lib/ruby_smb/client/negotiation.rb

```diff
@@ -24,6 +24,17 @@ def negotiate
         when '0x0311'
           parse_smb3_encryption_data(request_packet, response_packet)
         end
+
+        if response_packet && %w{0x0202 0x0210 0x0300 0x0302 0x0311 0x02ff}.include?(@dialect)
+          @server_guid = response_packet.server_guid
+          if response_packet.server_start_time != 0
+            @server_start_time = response_packet.server_start_time.to_time
+          end
+          if response_packet.system_time != 0
+            @server_system_time = response_packet.system_time.to_time
+          end
+        end
```

Comment:
Would it be possible to add this logic to the `parse_negotiate_response` method?
I believe this can be added under the `when RubySMB::SMB2::Packet::NegotiateResponse` case directly, which indirectly check the SMB version (no need to check the dialect in this case).